addons/mrp/models/mrp_production.py

```diff
@@ -658,17 +658,19 @@ def _onchange_producing(self):
 
     @api.onchange('lot_producing_id')
     def _onchange_lot_producing(self):
-        if self.product_id.tracking == 'serial':
-            if self.env['stock.move.line'].search_count([
-                ('company_id', '=', self.company_id.id),
-                ('product_id', '=', self.product_id.id),
-                ('lot_id', '=', self.lot_producing_id.id),
-                ('state', '!=', 'cancel')
-            ]):
+        if self.product_id.tracking == 'serial' and self.lot_producing_id:
+            quants = self.env['stock.quant'].search([('product_id', '=', self.product_id.id),
+                                                     ('lot_id', '=', self.lot_producing_id.id),
+                                                     ('quantity', '!=', 0),
+                                                     '|', ('location_id.usage', '=', 'customer'),
```

Comment:
Why only customer? Are you sure we should raise a warning if the user did the operation in the wrong order (deliver before encoding the production)?

Reply:
We are checking both 'customer' and 'internal' + 'transit' locations. I added in the customer location check because if a user reuses a SN that has already been sent to a customer, then there will be a validation error when they try to send that SN again to a customer without any prior warning messages. I figure an ignored warning is better than no warning. 

Also, to make it clearer for the cases where a SN is delivered before its MO is validated or its incoming picking is validated, I added an extra paragraph in the fixup to explain the possible case of assigning a SN when one already exists in one of these locations (similar to the one in the inventory check).  What do you think of that? If it's still not satisfactory, I can switch it back to ignoring customer quants.